### ISSUE TITLE
Can't build 0.9.0 from source tarball

### ISSUE THREAD

**Issue by lgbaldoni** [uim/sigscheme#7]
See attached [log](https://github.com/uim/sigscheme/files/1998892/sigscheme_log.txt).

**Comment by jpellegrini** [comment]
It also happens with the sources from git (pulled today, last commit was `a5516eb9a15539d2bf5e3b498d26f3f80c8fc906`), and without the `-j8` option to `make`.

**Comment by luigino** [comment]
@jpellegrini most of them don't apply and judging by the contents, debian seems to be using a modified source tarball.

**Comment by luigino** [comment]
@jpellegrini I see now debian is using the release tarball and not the source one.
Hence the report.

**Comment by jpellegrini** [comment]
Hello; I have tried again compiling a git checkout, and it fails, not for a different reason.

There is no `configure`  in the repository, so I ran

```
./autogen.sh
./configure
make
```

`autogen.sh` and `configure` succeed (return zero), but `make` fails:

```
$ make                                                                                                                                                                                                                                             
Making all in doc
make[1]: Entering directory '/home/jeronimo/pkg/scheme/sigscheme/doc'
make[1]: *** No rule to make target 'index.html', needed by 'all-am'.  Stop.
make[1]: Leaving directory '/home/jeronimo/pkg/scheme/sigscheme/doc'
make: *** [Makefile:544: all-recursive] Error 1
$
```

So, I tried removing `doc/Makefile` and inits place I put this file:

```
all:
        echo "nope"
```

and tried again. But `make` failed again, this time in a target that I believe I can't skip:

```
$ make
Making all in doc
make[1]: Entering directory '/home/jeronimo/pkg/scheme/sigscheme/doc'
echo "nope"
nope
make[1]: Leaving directory '/home/jeronimo/pkg/scheme/sigscheme/doc'
Making all in m4
make[1]: Entering directory '/home/jeronimo/pkg/scheme/sigscheme/m4'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory '/home/jeronimo/pkg/scheme/sigscheme/m4'
Making all in tools
make[1]: Entering directory '/home/jeronimo/pkg/scheme/sigscheme/tools'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory '/home/jeronimo/pkg/scheme/sigscheme/tools'
Making all in libgcroots
make[1]: Entering directory '/home/jeronimo/pkg/scheme/sigscheme/libgcroots'
make[1]: *** No rule to make target 'all'.  Stop.
make[1]: Leaving directory '/home/jeronimo/pkg/scheme/sigscheme/libgcroots'
make: *** [Makefile:544: all-recursive] Error 1
```

I have tried applying the pending pull requests, but didn't change anything.

**Comment by lgbaldoni** [comment]
[This](https://github.com/uim/sigscheme/files/5114542/sigscheme-090.txt) is how it fails now.

**Comment by kou** [comment]
Could you show how did you build to reproduce the case?

**Comment by lgbaldoni** [comment]
Same problem with 0.9.1, see [build log](https://github.com/uim/sigscheme/files/6356639/sigscheme-0.9.1_openSUSE_TW.txt)

**Comment by kou** [comment]
Why did you specify `--enable-maintainer-mode`?

**Comment by lgbaldoni** [comment]
> Why did you specify --enable-maintainer-mode?

I followed the advice you gave in https://github.com/uim/sigscheme/issues/7#issuecomment-678700922.

Otherwise I end up again with
```
Making all in doc
make[1]: *** No rule to make target 'index.html', needed by 'all-am'.  Stop.
make: *** [Makefile:545: all-recursive] Error 1
error: Bad exit status from /var/tmp/rpm-tmp.XYSVP8 (%build)
```

See [build log](https://github.com/uim/sigscheme/files/6357835/sigscheme-0.9.1_docfail.txt).

**Comment by kou** [comment]
OK.
You don't need to use `--enable-maintainer-mode` with the released archive: https://github.com/uim/sigscheme/releases/download/0.9.1/sigscheme-0.9.1.tar.gz
You need to use `--enable-maintainer-mode` with source checked out by `git`.

Did you use the released archive?
(You also don't need to run `autogen.sh` with the released archive.)

**Comment by lgbaldoni** [comment]
> Did you use the released archive?

No, I used the source tarball https://github.com/uim/sigscheme/archive/refs/tags/0.9.1.tar.gz.

Does it mean it's non functional?

**Comment by kou** [comment]
The source table https://github.com/uim/sigscheme/archive/refs/tags/0.9.1.tar.gz is same as cloning by `git`.
It's not prepared for release. You need to prepare manually by running `autogen.sh` and so on.

You should use https://github.com/uim/sigscheme/releases/download/0.9.1/sigscheme-0.9.1.tar.gz not `../refs/tags/...`.

**Comment by lgbaldoni** [comment]
Thing is, I did run `autogen.sh`. What else do I need to do?

**Comment by kou** [comment]
Why did you use https://github.com/uim/sigscheme/archive/refs/tags/0.9.1.tar.gz ?

**Comment by lgbaldoni** [comment]
I've just compared the contents of the two: I could not find binaries of any sort in the release tarball, so I can use it to build a rpm for distro inclusion.
It's not always the case, hence why I usually build from source.
